js/visearch.js

```diff
@@ -122,6 +116,43 @@
     }
   };
 
+  /**
+   * Get query parameters from url [URI] object
+   */
+  function getQueryParamValue(uri, name) {
+    return find([`__vi_${name}`, name], () => uri.getQueryParamValue(name));
+  }
+
+  /**
+   * Wrapper for callback function with additional send result_load event
+   */
+  function callbackWrap(callback, args) {
+    callback(args);
+
+    const curUri = new URI(window.location.href);
+    const reqid = getQueryParamValue(curUri, QUERY_REQID);
+    const imName = getQueryParamValue(curUri, QUERY_IMNAME);
+
+    // send out event if the pixel is in place
+    if (reqid && imName && context.vsPlacementLoaded[settings.placement_id]) {
```

Comment:
there are 2 scenarios:
- if it is SBI, we send result_load with only queryId
- if it is VSR, we send result_load with queryId and pid

Reply:
we also suppose to send only if the API is successful